lib/config_upgrader.js

```diff
@@ -235,8 +236,9 @@ class ConfigUpgrader {
       const config = new Config({
         configFile: configFile,
         cwd: process.cwd()
-      });
-      upgrader.upgrade(config.useSpecificConfig);
+      },
+      new Linter());
```

Comment:
Gordon told me that we can skip it. Release without upgrader in a separate channel for people who really want to use it and prepared to manually upgrade their configs. Add it later when we're ready to promote it to `latest`.

But I'm not sure about that now. ESLint 4 is much stricter about configs. It validates all keys, throws on anything unknown/dropped in previous versions. So maybe it's more urgent than previously thought.

 There's another bug to fix (plugin loading) before this can be considered working with a perfect config, but it would still fail integration tests because they all have invalid configs fro ESLint 4.